targets/arm/mikroe/sam/src/i2c/implementations/implementation_1/hal_ll_i2c_master.c

```diff
@@ -0,0 +1,855 @@
+/*!< @brief Helper macro for getting module specific control register structure base address // first register address */
+#define hal_ll_i2c_get_handle (hal_ll_i2c_master_handle_register_t *)hal_ll_i2c_get_module_state_address->hal_ll_i2c_master_handle
+/*!< @brief Helper macro for getting module specific control register structure */
+#define hal_ll_i2c_get_base_struct(_handle) ((hal_ll_i2c_base_handle_t *)_handle)
+/*!< @brief Helper macro for getting module specific base address directly from HAL layer handle */
+#define hal_ll_i2c_get_base_from_hal_handle ((hal_ll_i2c_hw_specifics_map_t *)((hal_ll_i2c_master_handle_register_t *)\
+                                            (((hal_ll_i2c_master_handle_register_t *)(handle))->hal_ll_i2c_master_handle))->hal_ll_i2c_master_handle)->base
+
+/*!< @brief Default I2C bit-rate if no speed is set */
+#define HAL_LL_I2C_MASTER_SPEED_100K                100000UL
+#define HAL_LL_I2C_MASTER_LOW_DIVIDER_SPEED_LIMIT   384000UL
+#define HAL_LL_I2C_MASTER_SPEED_400K                400000UL
+
+/*!< @brief Default pass count value upon reset */
+#define HAL_LL_I2C_DEFAULT_PASS_COUNT 0xFFFFU
+
+/*!< @brief Macro used for timeout */
+#define HAL_LL_I2C_LOOP_INDEX 10000U
+
+/*!< @brief Macros defining register bit location */
+#define HAL_LL_TWIHS_SR_TXCOMP_BIT      0
+#define HAL_LL_TWIHS_SR_RXRDY_BIT       1
+#define HAL_LL_TWIHS_SR_TXRDY_BIT       2
+#define HAL_LL_TWIHS_SR_ARBLST_BIT      9
+
+#define HAL_LL_TWIHS_CR_START_BIT       0
+#define HAL_LL_TWIHS_CR_STOP_BIT        1
+#define HAL_LL_TWIHS_CR_MSEN_BIT        2
+#define HAL_LL_TWIHS_CR_SVDIS_BIT       5
+#define HAL_LL_TWIHS_CR_SWRST_BIT       7
+
+#define HAL_LL_TWIHS_MMR_MREAD_BIT      12
+#define HAL_LL_TWIHS_MMR_DADR_START_BIT 16
+
+#define HAL_LL_PID_TWIHS2_BIT           9
+#define HAL_LL_PID_TWIHS0_BIT           19
+#define HAL_LL_PID_TWIHS1_BIT           20
+
+/*!< @brief Macros for speed calculation */
+#define HAL_LL_TWIHS_CWGR_CLKDIV_MAX    7
+
+// -------------------------------------------------------------- PRIVATE TYPES
+/*!< @brief I2C register structure */
+typedef struct
+{
+    hal_ll_base_addr_t cr;
+    hal_ll_base_addr_t mmr;
+    hal_ll_base_addr_t smr;
+    hal_ll_base_addr_t iadr;
+    hal_ll_base_addr_t cwgr;
+    hal_ll_base_addr_t _unused[3];
+    hal_ll_base_addr_t sr;
+    hal_ll_base_addr_t ier;
+    hal_ll_base_addr_t idr;
+    hal_ll_base_addr_t imr;
+    hal_ll_base_addr_t rhr;
+    hal_ll_base_addr_t thr;
+    hal_ll_base_addr_t smbtr;
+} hal_ll_i2c_base_handle_t;
+
+/*!< @brief I2C hw specific structure */
+typedef struct
+{
+    hal_ll_base_addr_t base;
+    hal_ll_pin_name_t module_index;
+    hal_ll_i2c_pins_t pins;
+    uint32_t speed;
+    uint8_t address;
+    uint16_t timeout;
+} hal_ll_i2c_hw_specifics_map_t;
+
+/*!< @brief I2C hw specific module values */
+typedef struct
+{
+    uint16_t pin_scl;
+    uint16_t pin_sda;
```

Comment:
`hal_ll_pin_name_t`

Reply:
Changed it for all source files where needed.